pkg/webhook/validation/uservalidation.go

```diff
@@ -48,10 +49,20 @@ func ValidateUserForFleetCRD(group string, namespacedName types.NamespacedName,
 // ValidateUserForResource checks to see if user is allowed to modify argued resource.
 func ValidateUserForResource(resKind string, namespacedName types.NamespacedName, whiteListedUsers []string, userInfo authenticationv1.UserInfo) admission.Response {
```

Comment:
Hi Arvind! I couldn't help noticing that the logic of two functions are essentially the same; is this expected?

Reply:
I wanted to add the node group check only for resources related to pod that's the only difference

Reply:
I see. The thing is, `system:nodes` is a group primarily reserved for kubelet and is considered to be owned by the system. If we would like to simplify things we can just greenlight this group completely.

Reply:
Besides, I have some doubt that kubelet will not touch APIs like deployment, daemon sets; they are very high-level and not of the concern for node-related components.